server/docs.js

```diff
@@ -66,12 +64,71 @@ async function fetchHTMLForId(id, resourceType, req, drive) {
     return fetchHTML(drive, id)
   }
 
-  const {data} = await drive.files.export({
-    fileId: id,
-    // text/html exports are not suupported for slideshows
-    mimeType: resourceType === 'presentation' ? 'text/plain' : 'text/html'
+  try {
+    const {data} = await drive.files.export({
+      fileId: id,
+      // text/html exports are not suupported for slideshows
+      mimeType: resourceType === 'presentation' ? 'text/plain' : 'text/html'
+    })
+
+    return data
+  } catch (e) {
+    // If the Google Drive API returns 403, we fall back to using the export link directly
+    if (e.response.status === 403) {
+      const manuallyFetchedData = await fetchManually(resourceType, exportLinks)
+      return manuallyFetchedData
+    } else {
+      console.log(e.code)
+      console.log(e.errors[0].message)
+      console.log('letting it bubble')
+    }
+  }
+}
+async function fetchManually(resourceType, exportLinks) {
+  const accessToken = await getGoogleAuth()
+
+  const axios = require('axios')
+  const exportLink = exportLinks['text/html']
+  const headers = {Authorization: `Bearer ${accessToken}`}
+
+  const fetchedData = await axios({
+    url: exportLink,
+    method: 'GET',
+    responseType: resourceType === 'presentation' ? 'text/plain' : 'text/html',
+    headers: headers
   })
-  return data
+    .then((response) => {
+      const fileContents = response.data
+      return fileContents
+    })
+    .catch((err) => {
+      console.error('Error downloading file:', err)
+    })
+  return fetchedData
+}
+
+async function getGoogleAuth() {
+  const {google} = require('googleapis')
```

Comment:
Is there some reason the require is scoped inside the function here? Now the dependency is required on every function call

Reply:
Not particularly, it should probably be somewhere else. I was wondering though, would it make sense to throw this whole function into the/an auth module? Not sure if this docs file is the right place for it 🤔